Services/Administration/OJS.Services.Administration.Business/Implementations/ProblemGroupsBusinessService.cs

```diff
@@ -153,21 +158,73 @@ await problemGroup.Problems
             await this.problemsData.SaveChanges();
 
             var newSubmissionTypeInSourceProblemsToAdd = new List<SubmissionTypeInProblem>();
+            var newResourcesToAdd = new List<ProblemResource>();
+            var newTestsToAdd = new List<Test>();
+
+            await this.GenerateNewTests(problem.Id, newTestsToAdd, currentNewProblem);
+
+            await this.GenerateNewResources(problem.Id, newResourcesToAdd, currentNewProblem);
 
             this.submissionTypesInProblemsData
-                .GetAllByProblem(problemId)
+                .GetAllByProblem(problem.Id)
                 .ForEach(stp =>
                     this.GenerateNewSubmissionTypesInProblem(stp, newSubmissionTypeInSourceProblemsToAdd, currentNewProblem));
 
+            currentNewProblem.Resources = newResourcesToAdd;
+            currentNewProblem.Tests = newTestsToAdd;
             currentNewProblem.SubmissionTypesInProblems = newSubmissionTypeInSourceProblemsToAdd;
             problemsToAdd.Add(currentNewProblem);
 
             this.problemsData.Update(currentNewProblem);
         }
 
+        private async Task GenerateNewTests(
+            int problemId,
+            ICollection<Test> newTestsToAdd,
+            Problem currentNewProblem)
+        {
+            var problemTests = await this.testsData
+                .GetAllByProblem(problemId)
+                .ToListAsync();
+
+            foreach (var test in problemTests)
+            {
+                var newTest = test.Map<Test>();
+                newTest.Id = 0;
+                newTest.ProblemId = currentNewProblem.Id;
+                newTest.Problem = currentNewProblem;
+
+                await this.testsData.Add(newTest);
+                await this.testsData.SaveChanges();
+                System.Console.WriteLine(newTest.Id);
+                newTestsToAdd.Add(newTest);
+            }
+        }
+
+        private async Task GenerateNewResources(
+            int problemId,
+            ICollection<ProblemResource> newResourcesToAdd,
+            Problem currentNewProblem)
+        {
+            var problemResources = await this.problemResourcesData
+                .GetByProblemQuery(problemId)
+                .ToListAsync();
```

Comment:
Call `.AsNoTracking()` and reuse the retrieved entities to generate new ones (just clear the Id)